### ISSUE TITLE
Syslog source documentation is incorrect about the `.timestamp` field definition

### ISSUE THREAD

**Issue by hhromic** [timberio/vector#7442]
### Vector Version

```
vector 0.13.1 (v0.13.1 x86_64-unknown-linux-gnu 2021-04-29)
```

### Vector Configuration File

```yaml
sources:
  syslog-source:
    type: syslog
    mode: udp
    address: 0.0.0.0:5555

sinks:
  console-sink:
    type: console
    inputs:
      - syslog-source
    encoding:
      codec: json
```

### Expected Behavior

The [Syslog source documentation](https://vector.dev/docs/reference/configuration/sources/syslog/#timestamp) states that:

> timestamp
> The exact time the event was ingested into Vector.

Therefore, if we send the following example syslog message to Vector:
```
<165>1 2003-10-11T22:14:15.003Z mymachine.example.com evntslog - ID47 [exampleSDID@32473 iut="3" eventSource="Application" eventID="1011"] BOMAn application event log entry...
```

According to the documentation, we should expect the `.timestamp` field to be the current time when Vector ingested the event, not the time embededd into the syslog message.

### Actual Behavior

The resulting field is populated with the syslog packet's timestamp:
```json
{"appname":"evntslog","exampleSDID@32473":{"eventID":"1011","eventSource":"Application","iut":"3"},"facility":"local4","host":"mymachine.example.com","hostname":"mymachine.example.com","message":"BOMAn application event log entry...","msgid":"ID47","severity":"notice","source_ip":"127.0.0.1","source_type":"syslog","timestamp":"2003-10-11T22:14:15.003Z","version":1}
```

**Important Note**

I believe that the current behaviour is more useful than providing the ingestion time (which can be easily obtained in VRL using the `now()` function). Therefore, I suggest to amend the documentation to reflect that the `.timestamp` field is actually extracted from the received syslog message (or the current time if the syslog message can't be parsed).

Just for completeness, I can confirm that the invalid case is working as expected in Vector:
```json
{"host":"127.0.0.1","message":"blabla","source_ip":"127.0.0.1","source_type":"syslog","timestamp":"2021-05-13T09:55:08.764803300Z"}
```
In that example I just sent the string `blabla` to Vector.

In summary, I think only the documentation needs to be corrected.

### Example Data

```
<165>1 2003-10-11T22:14:15.003Z mymachine.example.com evntslog - ID47 [exampleSDID@32473 iut="3" eventSource="Application" eventID="1011"] BOMAn application event log entry...
```

**Comment by hhromic** [comment]
Tried to edit the documentation myself from the "Edit this page" link, but the link is broken:
https://github.com/timberio/vector/edit/master/docs/reference/configuration/sources/syslog.cue

Looks like the links should be instead:
https://github.com/timberio/vector/edit/master/docs/reference/components/sources/syslog.cue

**Comment by jszwedko** [comment]
Thanks @hhromic . I'll see about getting the edit links fixed too.
